core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2253,13 +2253,38 @@ private[spark] object Utils extends Logging {
   def isDynamicAllocationEnabled(conf: SparkConf): Boolean = {
     val numExecutor = conf.getInt("spark.executor.instances", 0)
     val dynamicAllocationEnabled = conf.getBoolean("spark.dynamicAllocation.enabled", false)
+    val overrideExecutors = conf.getBoolean("spark.dynamicAllocation.overrideNumInstances", false)
```

Comment:
I think this just piles onto the confusion though, with yet another flag to control flags. I think this is a documentation issue, and it already generates warnings when you set these things together. I personally favor leaving it as is.